src/transport/concentration_model.hh

```diff
@@ -45,8 +45,24 @@ public:
 	class ModelEqData : public TransportBase::TransportEqData {
 	public:
 
-		/// Boundary conditions (Dirichlet) for concentrations.
-		BCField<3, FieldValue<3>::Vector> bc_conc;
+		enum Concentration_bc_types {
+			bc_inflow,
+			bc_dirichlet,
+			bc_total_flux,
+			bc_diffusive_flux
+		};
+
+		/// Convert bc_type in model to bc_type in abstract setting.
+		static const std::map<unsigned int,unsigned int> bc_type_conversion;
```

Comment:
I think we can remove this map in order to don't be tempted to use it 
in get_bc_type. I prefer to use an lookup vector instead .